--- /home/alt/git-repo/script-python3/downloader-booru/classes/database.py	2019-10-15 10:24:17.969773813 +0900
+++ database.py	2021-02-15 19:31:03.419532718 +0900
@@ -7,34 +7,16 @@
 from helpers.misc import print_log
 
 class checks():
-    def __init__(self, database_sqlx, **options):
-        self.database_object = database_sqlx
+    def __init__(self, database_obj, **options):
+        self.database_object = database_obj
         self.database_verbose = options.pop('verbose', False)
 
-    def run(self, **database_data):
-        #global self.database_object
-        db_table = database_data.pop('table', None)
-        db_amode = database_data.pop('mode', None)
-        if self.database_object is not None \
-        and db_table is not None:
-            if db_table is not None and len(db_table) > 0:
-                return self.database_object.run(
-                    mode  = db_amode, 
-                    table = db_table, 
-                    **database_data)
-        else:
-            return None
-
     def view(self, view_href, view_table, **options):
         check_verbose = options.pop('verbose', self.database_verbose)
-        check_result = namedtuple(
-            'DbCheckResultSingle', ['skip', 'data', 'count'])
+        check_result = namedtuple('DbCheckResultSingle', ['skip', 'data', 'count'])
         if self.database_object is not None:
             view_id = options.pop('id', gen_viewid(view_href))
-            view_data = self.run(
-                mode  = 'select row', 
-                table = view_table, 
-                view  = view_id)
+            view_data = self.database_object.select_from(table=view_table, view=view_id)
             view_dlen = len(view_data) if view_data is not None else 0
             if check_verbose:
                 print_log('info', 'CHECK [VSGL] - Table: "%s", Url: "%s", Rows: [%s]', view_table, view_href, view_dlen)
@@ -54,20 +36,14 @@
 
     def views(self, index_views, index_table, **options):
         check_verbose = options.pop('verbose', self.database_verbose)
-        check_results = namedtuple(
-            'DbCheckResultMulti', ['skip', 'data', 'count'])
+        check_results = namedtuple('DbCheckResultMulti', ['skip', 'data', 'count'])
         if self.database_object is not None:
             check_count = 0
             check_store = []
             for view_href in index_views:
                 view_id = gen_viewid(view_href)
                 # print(view_id)
-                check_single = self.view(
-                    view_href, 
-                    index_table, 
-                    id=view_id, 
-                    verbose=check_verbose
-                )
+                check_single = self.view(view_href, index_table, id=view_id, verbose=check_verbose)
                 if check_single.skip:
                     check_count += 1
                     check_store.append(check_single.skip)
@@ -83,10 +59,6 @@
                 )
             else:
                 pass
-            return check_results(
-                skip = check_skip, 
-                data = check_store, 
-                count= check_count
-            )
+            return check_results(skip=check_skip, data=check_store, count=check_count)
         else:
-            return check_results(skip = False, data = None, count = None)
+            return check_results(skip=False, data=None, count=0)
